sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -1110,6 +1112,14 @@ class DataFrameAggregateSuite extends QueryTest
     val e = intercept[AnalysisException](arrayDF.groupBy(struct($"col.a")).count())
     assert(e.message.contains("requires integral type"))
   }
+
+  test("SPARK-34716: Support ANSI SQL intervals by the aggregate function `sum`") {
```

Comment:
Could you test more cases:
1. Negative tests such as overflow
2. Aggregate nulls and non-nulls
3. Looking at https://github.com/apache/spark/pull/26325, we will need to add more tests as soon as we support construction of the intervals in SQL via cast or `make_interval`

Reply:
For your first suggestion. I think we should not add the negative tests. Because we have the negative tests for `Add`. Sum(int), Sum(decimal) not add the negative tests, because they depend the robust of `Add`.

Reply:
For the third suggestion, If we support intervals in SQL, we will add new test cases.